app/assets/javascripts/angular/controllers/graph_ctrl.js

```diff
@@ -143,7 +147,18 @@ angular.module("Prometheus.controllers").controller('GraphCtrl',
     $scope.addAxis();
   }
 
+  $scope.$watch('graph.range', $scope.refreshGraph);
+  $scope.$watch('graph.endTime', $scope.refreshGraph);
```

Comment:
This doesn't seem to fire for the end time, neither when clicking on the next/prev buttons, nor when selecting a time via the datapicker. It does update when I click the "x" clear button though. Can you reproduce?